### PR TITLE
samples: Pub/Sub Spring

### PR DESCRIPTION
**Pull request by anguillanneuf**
Add code snippets for a Pub/Sub Spring user guide (similar to how they are done for [Spanner Spring](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/spanner/spring-data)). 

Things worth mentioning:
- ~~`shared-configuration` is not added as a parent in `pom.xml` because `spring-boot-starter-parent`need to exist as a parent.~~
- No test are added because these will be used as code snippets. I also don't see tests in the Spanner samples.

### CODE REVIEW

File: pubsub/spring/src/main/java/demo/PubSubApplication.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        Flux.fromStream(
+                // Generate a stream that sends a numbered message every 10 seconds.
+                Stream.generate(
+                    new Supplier<Message<String>>() {
+                      @Override
+                      public Message<String> get() {
+                        try {
+                          Thread.sleep(10000);
+                        } finally {
+                          Random rand = new Random();
```
Comment (lesv) [inline pubsub/spring/src/main/java/demo/PubSubApplication.java:135]: Ok, but shouldn't this be somewhere earlier - possibly as a classvar or static so it gets reused?

Comment (anguillanneuf) [reply pubsub/spring/src/main/java/demo/PubSubApplication.java]: Sounds good. Updated.

**Comment by anguillanneuf** [comment]
Thanks @lesv for the review! I modified the sample as suggested.

File: pubsub/spring/src/main/java/demo/PubSubApplication.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+@SpringBootApplication
+public class PubSubApplication {
+
+  private static final Log LOGGER = LogFactory.getLog(PubSubApplication.class);
+
+  public static Message<String> generateOneMessagePeriodically(int millis) {
+    try {
+      Thread.sleep(millis);
+    } finally {
+      Random rand = new Random();
```
Comment (lesv) [inline pubsub/spring/src/main/java/demo/PubSubApplication.java:55]: LOL - I was meaning consider making `rand` static or a class variable.  https://docs.oracle.com/javase/8/docs/api/java/util/Random.html

It's a bit heavyweight.

Comment (anguillanneuf) [reply pubsub/spring/src/main/java/demo/PubSubApplication.java]: Oh sorry I misunderstood!

**Comment by anguillanneuf** [comment]
@lesv I misunderstood your earlier comment. I'm also not authorized to merge.

**Pull request merged**